mempool.go

```diff
@@ -799,10 +800,9 @@ func (mp *txMemPool) FetchTransaction(txHash *btcwire.ShaHash) (*btcutil.Tx, err
 // more details.
 //
 // This function MUST be called with the mempool lock held (for writes).
-func (mp *txMemPool) maybeAcceptTransaction(tx *btcutil.Tx, isOrphan *bool, isNew, rateLimit bool) error {
-	if isOrphan != nil {
-		*isOrphan = false
-	}
+func (mp *txMemPool) maybeAcceptTransaction(tx *btcutil.Tx,
+	isNew, rateLimit bool) (missingParents []*btcwire.ShaHash, err error) {
```

Comment:
I did notice that it was not doing bare returns and agree that they're the culprit, but I don't even want to open the door for their usage at all.  If they aren't even in the declaration, there is no way to inadvertently forget to return specific values in one of the error paths.  The documentation is easy enough to put in the comment.